cairo/ethereum/cancun/trie.cairo

```diff
@@ -514,6 +561,42 @@ func trie_set_TrieTupleAddressBytes32U256{
     return ();
 }
 
+func trie_set_TrieBytesOptionalUnionBytesLegacyTransaction{
+    poseidon_ptr: PoseidonBuiltin*, trie: TrieBytesOptionalUnionBytesLegacyTransaction
+}(key: Bytes, value: UnionBytesLegacyTransaction) {
+    alloc_locals;
+    let dict_ptr = cast(trie.value._data.value.dict_ptr, DictAccess*);
+
+    local is_default;
+    if (value.value == trie.value.default.value) {
```

Comment:
is this equality coherent?
```
struct UnionBytesLegacyTransactionEnum {
    bytes: Bytes,
    legacy_transaction: LegacyTransaction,
}

struct UnionBytesLegacyTransaction {
    value: UnionBytesLegacyTransactionEnum*,
}
```

Reply:
You might need a UnionBytesLegacyTransaction__eq__

Reply:
Or easier: `if (cast(value.value, felt) == 0)`

Reply:
we're comparing pointer-to-pointer. for me this is the right way of doing. this is not opinionated about what the default value is

Reply:
you're right this is wrong

Reply:
Do you have a preference @enitrat @obatirou

Reply:
let's compare to None, we don't care about genericity

Reply:
[7e9e3a6](https://github.com/kkrt-labs/keth/pull/645/commits/7e9e3a6971bc0045b05ab463a18d3884f4dd0c26)